blog/03-08-2023-Build-A-Full-Stack-Decentralized-Fund-Raising-Dapp-On-The-Celo-Blockchain/Build-A-Full-Stack-Decentralized-Fund-Raising-Dapp-On-The-Celo-Blockchain.md

```diff
@@ -0,0 +1,666 @@
+
+
+
+```javascript
+ const addCampaign = async (_image, _description, _beneficiary, _goal) => {
+    try {
+      await contract.methods
+        .createCampaign(_image, _description, _beneficiary, _goal)
+        .send({ from: address });
+      getCampaigns();
+    } catch (error) {
+      alert(error);
+    }
+  };
+```
+After that, we have the `addCampaign` function, which takes in the required parameters to create a new campaign and sends a transaction using the `createCampaign` method of the contract object. It then calls the `getCampaigns` function to update the list of campaigns.
+
+```javascript
+ const fundCampaign = async (_index, _ammount) => {
+    try {
+      const cUSDContract = new kit.web3.eth.Contract(IERC, cUSDContractAddress);
+
+      await cUSDContract.methods
+        .approve(contractAddress, _ammount)
+        .send({ from: address });
+      await contract.methods
+        .fundCampaign(_index, _ammount)
+        .send({ from: address });
+      getCampaigns();
+      getBalance();
+      alert("you have successfully sent cusd to this user");
+    } catch (error) {
+      alert(error);
+    }
+  };
+```
+Lastly, we have the `fundCampaign` function which takes in the index of the campaign and the amount of cUSD to be sent. It first approves the contract to spend the specified amount of cUSD using the approve method of the `cUSDContract` object. It then sends a transaction to the contract to fund the `campaign` using the `fundCampaign` method. Finally, it calls the `getCampaigns` and `getBalance` functions to update the list of `campaigns` and the user's `cUSD` balance.
+
+```javascript
+useEffect(() => {
+connectToWallet();
+}, []);
+
+useEffect(() => {
+if (kit && address) {
+getBalance();
+}
+}, [kit, address, getBalance]);
+
+useEffect(() => {
+if (contract) {
+getCampaigns();
+}
+}, [contract, getCampaigns]);
+
+```
+We use useEffect to call the connectToWallet, getBalance, and getCampaigns functions at the appropriate times.
+
+```javascript
+return (
+    <div className="App">
+      <Home cUSDBalance={cUSDBalance} addCampaign={addCampaign} />
+      <Campaigns
+        campaigns={campaigns}
+        fundCampaign={fundCampaign}
+        walletAddress={address}
+      />
+    </div>
+  );
+}
+
+export default App;
+
+```
+And finally, we render the App component and return the Home and Campaigns components with the necessary props.
+
+## Next Steps
+
+I hope you learned a lot from this tutorial. Here are some relevant links that would aid your learning further.
+
+- <https://docs.celo.org/>
```

Comment:
can you embed the links here like you've done throughout the tutorial 🙃

Reply:
it's important for readers to see the links they are going to if they want to learn more about a specific topic

Reply:
No seeing the link is not what's important, taking the reader to the right reference is what is
That is why you should embed it with the right description or example:

the first link would rather look like this
[celo docs](https://docs.celo.org/)